chains/solana/contracts/programs/fee-quoter/src/instructions/v1/messages.rs

```diff
@@ -134,18 +123,30 @@ pub fn process_extra_args(
                 message_contains_tokens,
             )?)
         }
+        _ => {
+            // Extra args are optional for non-SVM destinations. In case there
+            // are extra args, they must be prefixed by a four byte tag
+            // -> bytes4(keccak256("CCIP EVMExtraArgsV2"));
+            let Some(tag) = extra_args.get(..4) else {
+                return Ok(ProcessedExtraArgs::defaults(dest_config));
+            };
 
-        _ => Err(FeeQuoterError::InvalidChainFamilySelector.into()),
+            let mut data = &extra_args[4..];
+            parse_and_validate_generic_extra_args(tag.try_into().unwrap(), &mut data)
```

Comment:
should we map this error and fail into a custom one?

Reply:
I don't think we need to: we've already verified that the tag exists, so all that remains is expectations after having read that tag. The errors are still relevant to the caller (arguments couldn't be processed because the data after the tag was wrong).